### DISCUSSION TITLE
Apex extension has issue with cpu & memory consumption

### DISCUSSION THREAD

**Discussion by gbockus-sf** [forcedotcom/salesforcedx-vscode#4440] [General]
This discussion is meant to track the ongoing issue of resource usage by the apex extension.  

## Root of the problem
The Language server used by the Apex extension is utilizing a package provided by the Apex team written in java that has the following resource usage on my MacOS laptop: 
Memory : 530 MB

This is problematic on systems that are memory constrained and result in the APEX extension failing to load. 

I'm starting this discussion to track the number of issues we have reported around this issue and to track potential solutions. 

## Troubleshooting
If you are receiving an "unresponsive" error, run the following commands in your project. The first will check to see *if* you have a stale Apex Language Server. Grab the process ID, and run it in the second command to reset it.

# Mac
`ps -eo pid,args | grep "[A]pexLanguageServerLauncher"` 
`kill <process_number>`

# Windows
`tasklist` -> find something like [A]pexLanguageServerLauncher
`taskkill /PID PID /F`

**Comment by gbockus-sf** [comment]
#4409

**Comment by gbockus-sf** [comment]
#4406

**Comment by randi274** [comment]
#4537
#4444
#4484

**Comment by gbockus-sf** [comment]
https://github.com/forcedotcom/salesforcedx-vscode/issues/4547

**Comment by gbockus-sf** [comment]
@peternhale Can we call this problem solved?
